src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/ExternalTaskWorker.java

```diff
@@ -48,20 +57,31 @@ public void setupClient() {
     }
 
     public void checkIdempotency(ExternalTask externalTask, ExternalTaskService externalTaskService) {
-        Boolean isDuplicate =  externalTask.getVariable("isDuplicate");
-        if (isDuplicate == null || isDuplicate) {
+
+
+        Optional<IdempotentKeys> existingTask = idempotentKeysRepository.findById(new IdempotentId(externalTask.getVariable("idempotentKey"),"ia"));
+        if (existingTask.isPresent()) {
             Map<String, Object> processVariables = singletonMap(
                 "isDuplicate",
-                false
+                true
             );
-
-            LOGGER.info("Duplicate was hit.");
-
+            LOGGER.info("It was present!!!!!");
```

Comment:
I love those exclamation marks, but we should probably improve that message.

Reply:
I see there is a Slf4J annotation, should this be removed too?